dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/handler/MetadataHandler.java

```diff
@@ -135,7 +135,7 @@ public void addMetaData(DataQueryParams params, Grid grid) {
 
       if (params.isShowHierarchy()) {
         Map<Object, List<?>> ancestorMap =
-            activeOrgUnits.stream()
```

Comment:
I'm wondering if would be safer to do it only for download/export endpoints. I'm afraid we are changing the existing behavior for the `isShowHierarchy` flag.

Reply:
Basically, we could check if the request originates from download endpoints.

Reply:
done